### PR TITLE
particl-core: 0.16.2.0 -> 0.17.0.2

### PR DESCRIPTION
**Pull request by demyanrogozhin**
###### Motivation for this change
Update version. Disabled tests and benchmartk to speed up Hydra build.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by xeji** [comment]
I agree that running benchmarks during Hydra build doesn't make much sense. But I suggest to keep the tests enabled unless they take so much time that Hydra builds time out. Is that the case here?

File: pkgs/applications/altcoins/particl/particl-core.nix [hunk 1]
```diff
@@ -29,12 +29,17 @@ stdenv.mkDerivation rec {
     unixtools.hexdump
   ];
 
-  configureFlags = [ "--with-boost-libdir=${boost.out}/lib" ];
+  configureFlags = [
+    "--disable-bench"
+    "--enable-tests=no"
```
Comment (Mic92) [inline pkgs/applications/altcoins/particl/particl-core.nix]: Did we run the tests before?

Comment (demyanrogozhin) [reply pkgs/applications/altcoins/particl/particl-core.nix]: I think this flag only to build tests, and it was never used (not sure about it).
I trust @tecnovert (particl-core developer) to run all tests before publishing release.

**Comment by demyanrogozhin** [comment]
> I agree that running benchmarks during Hydra build doesn't make much sense. But I suggest to keep the tests enabled unless they take so much time that Hydra builds time out. Is that the case here?

Yes, it takes time. I trying to reduce build time as much as possible, because in parallel PR for particl-qt all builds timed out (even on x86_64): #46760

I have no clue how to speed-up compilation or prevent Hydra from timeout...
Kindly appreciate your advice)

**Comment by xeji** [comment]
ofborg timeouts (like you see in #46760)  and hydra timeouts are two different things. Most builds that time out on ofborg will succeed on hydra, so usually if a build times out on ofborg we merge anyway and see what happens on hydra.  I do not see any recent _hydra_ timeouts for `particl-core`, so there's no urgent need to speed it up.

But what you can try is `enableParallelBuilding = true;`. It doesn't work with all packages, but if it does, it speeds up the build by using all available CPU cores (currently this is only built on one core).

File: pkgs/applications/altcoins/particl/particl-core.nix (hunk 1 above)
Comment (demyanrogozhin) [reply pkgs/applications/altcoins/particl/particl-core.nix]: I added `doCheck` that run tests, also running tests required python3 which was added: 233dad81ebc164520fb763cbb47a16c88655888d

**Comment by demyanrogozhin** [comment]
> But what you can try is `enableParallelBuilding = true;`. It doesn't work with all packages, but if it does, it speeds up the build by using all available CPU cores (currently this is only built on one core).
Didn't know that, thanks for clarification @xeji. Added this tweak and enable tests.

**Comment by xeji** [comment]
@GrahamcOfBorg build particl-core

**Pull request merged**